diff --git i/src/rewrites.rs w/src/rewrites.rs
index b2a6484..1f21bc5 100644
--- i/src/rewrites.rs
+++ w/src/rewrites.rs
@@ -39,7 +39,11 @@ pub(crate) fn pre_process<T>(
         &matched.destination,
         &matched.replacer,
     ) {
-        Ok(dest) => dest,
+        Ok(dest) => if req.uri().query().is_some() {
+            dest + "?" + req.uri().query().unwrap()
+        } else {
+            dest
+        },
         Err(err) => return handle_error(err, opts, req),
     };
 
