diff --git a/src/rewrites.rs b/src/rewrites.rs
index e128059..dc5b88a 100644
--- a/src/rewrites.rs
+++ b/src/rewrites.rs
@@ -26,7 +26,11 @@ pub(crate) fn pre_process<T>(
 
     let matched = rewrite_uri_path(uri_path, Some(rewrites))?;
     let dest = match replace_placeholders(uri_path, &matched.source, &matched.destination) {
-        Ok(dest) => dest,
+        Ok(dest) => if req.uri().query().is_some() {
+            dest + "?" + req.uri().query().unwrap()
+        } else {
+            dest
+        },
         Err(err) => return handle_error(err, opts, req),
     };
 
