--- a/pkg/build/slsa.go
+++ b/pkg/build/slsa.go
@@ -29,8 +29,8 @@
 const (
 	intotoStatementType         = "https://in-toto.io/Statement/v1"
 	slsaProvenanceStatementType = "https://slsa.dev/provenance/v1"
-	melangeBuilder              = "https://chainguard.dev/prod/builders/melange/v1"
-	melangeBuildType            = "https://chainguard.dev/buildtypes/melange/v1"
+	melangeBuilder              = "https://g5d.dev/containers/builders/melange/v1"
+	melangeBuildType            = "https://g5d.dev/containers/buildtypes/melange/v1"
 )
 
 func (pc *PackageBuild) generateSLSA() ([]byte, error) {
