diff --git i/pkg/config/config.go w/pkg/config/config.go
index 4d1326f..b227287 100644
--- i/pkg/config/config.go
+++ w/pkg/config/config.go
@@ -840,6 +840,7 @@ func (p Pipeline) SBOMPackageForUpstreamSource(licenseDeclared, supplier string,
 			IDComponents:     idComponents,
 			Name:             pkgName,
 			Version:          pkgVersion,
+			LicenseDeclared:  licenseDeclared,
 			Namespace:        supplier,
 			Checksums:        checksums,
 			PURL:             pu,
