--- a/registry/registry.go	2025-11-02 10:23:28.809116185 +0000
+++ b/registry/registry.go	2025-11-02 10:32:37.746277681 +0000
@@ -18,7 +18,6 @@
 	gorhandlers "github.com/gorilla/handlers"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
-	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 	"golang.org/x/crypto/acme"
 	"golang.org/x/crypto/acme/autocert"
 	"golang.org/x/net/http2"
@@ -29,7 +28,6 @@
 	"github.com/distribution/distribution/v3/internal/dcontext"
 	"github.com/distribution/distribution/v3/registry/handlers"
 	"github.com/distribution/distribution/v3/registry/listener"
-	"github.com/distribution/distribution/v3/tracing"
 	"github.com/distribution/distribution/v3/version"
 )
 
@@ -163,14 +161,9 @@
 		handler = applyHandlerMiddleware(config, handler)
 	}
 
-	err = tracing.InitOpenTelemetry(app.Context)
-	if err != nil {
-		return nil, fmt.Errorf("error during open telemetry initialization: %v", err)
-	}
 	if config.HTTP.H2C.Enabled {
 		handler = h2c.NewHandler(handler, &http2.Server{})
 	}
-	handler = otelHandler(handler)
 
 	server := &http.Server{
 		Handler: handler,
@@ -184,13 +177,6 @@
 	}, nil
 }
 
-// otelHandler returns an http.Handler that wraps the provided `next` handler with OpenTelemetry instrumentation.
-// This instrumentation tracks each HTTP request, creating spans with names derived from the request method and URL path.
-func otelHandler(next http.Handler) http.Handler {
-	return otelhttp.NewHandler(next, "",
-		otelhttp.WithSpanNameFormatter(func(_ string, r *http.Request) string { return r.Method + " " + r.URL.Path }))
-}
-
 // takes a list of cipher suites and converts it to a list of respective tls constants
 // if an empty list is provided, then the defaults will be used
 func getCipherSuites(names []string) ([]uint16, error) {
