--- a/pkg/build/build_implementation.go
+++ b/pkg/build/build_implementation.go
@@ -20,7 +20,6 @@
 	"context"
 	"crypto/sha256"
 	"encoding/hex"
-	"encoding/json"
 	"fmt"
 	"hash"
 	"io"
@@ -274,10 +273,6 @@
 		}
 	}
 
-	if err := bc.WriteEtcApkoConfig(ctx); err != nil {
-		return nil, fmt.Errorf("failed to install apko config: %w", err)
-	}
-
 	if err := mutatePaths(bc.fs, &bc.o, &bc.ic); err != nil {
 		return nil, fmt.Errorf("failed to mutate paths: %w", err)
 	}
@@ -367,24 +362,6 @@
 	return nil
 }
 
-func (bc *Context) WriteEtcApkoConfig(_ context.Context) error {
-	// Encode the image configuration and write it to /etc/apko.json
-	f, err := bc.fs.Create("/etc/apko.json")
-	if err != nil {
-		return fmt.Errorf("creating /etc/apko.json: %w", err)
-	}
-	if err := json.NewEncoder(f).Encode(bc.ic); err != nil {
-		return fmt.Errorf("encoding image config: %w", err)
-	}
-	if err := f.Close(); err != nil {
-		return fmt.Errorf("closing /etc/apko.json: %w", err)
-	}
-	if err := bc.fs.Chmod("/etc/apko.json", 0444); err != nil {
-		return fmt.Errorf("chmod /etc/apko.json: %w", err)
-	}
-	return nil
-}
-
 // WriteIndex saves the index file from the given image configuration.
 func WriteIndex(ctx context.Context, o *options.Options, idx v1.ImageIndex) (string, error) {
 	log := clog.FromContext(ctx)
